### PR TITLE
feat: artist series sticky filter

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves []

### Description
[Initial implementation](https://github.com/artsy/eigen/pull/4705)

#### Demo
https://user-images.githubusercontent.com/3513494/121366762-684f8b00-c942-11eb-9ade-d47ebf5ba1c0.mp4

### PR Checklist (tick all before merging)

- [ ] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [ ] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [ ] I have documented any follow-up work that this PR will require, or it does not require any.
- [ ] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [ ] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

### CODE REVIEW

File: src/lib/Scenes/ArtistSeries/ArtistSeries.tsx [hunk 1]
```diff
@@ -84,66 +70,67 @@ export const ArtistSeries: React.FC<ArtistSeriesProps> = (props) => {
...
+                    currentArtistSeriesExcluded
+                  />
+                </Box>
+              </>
+            ) : undefined}
+            stickyHeaderContent={(
+              <Box backgroundColor="white100" py={1}>
+                <Flex flexDirection="row" px={2} justifyContent="space-between" alignItems="center">
+                  <Text variant="subtitle" color="black60">
+                    Showing 100 works
```
Comment (mdole) [inline src/lib/Scenes/ArtistSeries/ArtistSeries.tsx:103]: We should update this hardcoded value before merge :)

File: src/lib/Components/StickyHeaderPage.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    Animated.cond(
+      Animated.greaterThan(scrollStickyHeaderOffsetY, footerOffsetY),
+      Animated.max(
+        Animated.sub(footerOffsetY, scrollStickyHeaderOffsetY),
+        Animated.multiply(-1, stickyHeaderHeight ?? 0)
+      ),
+      stickyHeaderOffsetY
+    ),
+    stickyHeaderOffsetY
+  )
```
Comment (mdole) [inline src/lib/Components/StickyHeaderPage.tsx:39]: @dimatretyak can you help me understand what this does? I'm guessing it's to allow the sticky header to scroll out of view if we have a large-enough footer, but I'm getting a bit lost trying to read it!

**Review by mdole — COMMENTED** [review]
@dimatretyak I think this is a really good approach! I like the way you created a pattern that could be used elsewhere in the app where we want a sticky header without tabs. I'll tag Pavlos for review as well because he's a more experienced mobile dev than I am, but what you have here is looking good to me.

**Comment by mdole** [comment]
@dimatretyak you mentioned via Slack that you were seeing some small lags when scrolling on Android, which you were also seeing with StickyTabPage. I don't see any lags when scrolling in the app on my device - can you perhaps upload a video of the issue? Does it happen on a physical device, on a simulator, or both?

File: src/lib/Components/StickyHeaderPage.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/lib/Components/StickyHeaderPage.tsx:39]: @mdole
line 29 – if the user passed the footer component, then we check whether the coordinate of the beginning of this component was read
Line 31 - if the user has started scrolling the footer area, then run a smooth translateY reduction for the sticky header (line 33) until the sticky header is completely hidden (lines 33-34)

Comment (dimatretyak) [reply src/lib/Components/StickyHeaderPage.tsx:39]: If I have time, I will try to refactor the component a little

File: src/lib/Scenes/ArtistSeries/ArtistSeries.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/lib/Scenes/ArtistSeries/ArtistSeries.tsx:103]: No problems. I've left hardcoded for the demo

**Comment by dimatretyak** [comment]
> @dimatretyak you mentioned via Slack that you were seeing some small lags when scrolling on Android, which you were also seeing with StickyTabPage. I don't see any lags when scrolling in the app on my device - can you perhaps upload a video of the issue? Does it happen on a physical device, on a simulator, or both?

@mdole  I tested it on a physical device. If you look closely, you may notice a slight twitching of the header while scrolling. Perhaps this problem will not be in the release build

https://user-images.githubusercontent.com/3513494/121727154-33833580-caf4-11eb-909d-fb0af0b2841e.mp4

File: src/lib/Components/StickyHeaderPage.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/lib/Components/StickyHeaderPage.tsx]: Made a small refactoring of the `StickyHeaderPage`

Comment (mdole) [reply src/lib/Components/StickyHeaderPage.tsx]: Thanks for the explanation and refactor, that helps!

**Review by mdole — APPROVED** [review]
@dimatretyak nice work, and thanks for the video demo! The code is looking good to me. I'm not sure what optimizations we do differently on Production builds and whether they would fix the scrolling animation. I'll start a thread in #dev-help and see if anyone else knows!

**Comment by dimatretyak** [comment]
@mdole @MounirDhahri unfortunately, the problem still remains

https://user-images.githubusercontent.com/3513494/123660257-2bd5c780-d83c-11eb-8157-d16fb2eca187.mp4

**Comment by mdole** [comment]
Yeah, that sounds good to me as well. It's not all that noticeable unless you're really looking for it

**Pull request merged**